polaris-core/src/main/java/org/apache/polaris/core/storage/StorageLocation.java

```diff
@@ -39,8 +40,8 @@ public static StorageLocation of(String location) {
   protected StorageLocation(@Nonnull String location) {
     if (location == null) {
       this.location = null;
-    } else if (location.startsWith("file:/") && !location.startsWith("file:///")) {
-      this.location = URI.create(location.replaceFirst("file:/+", "file:///")).toString();
+    } else if (location.startsWith("file:/") && !location.startsWith(LOCAL_PATH_PREFIX)) {
+      this.location = URI.create(location.replaceFirst("file:/+", LOCAL_PATH_PREFIX)).toString();
```

Comment:
Can we normalized the location here so that no change is needed in method `isChildOf()`?  We can do that as the following:
```
} else if (location.startsWith("/") ) {
      this.location = URI.create(location.replaceFirst("/+", LOCAL_PATH_PREFIX)).toString();
```